File: Python/Math-Array-String/06.py
# 3028. Ant on the Boundary

from typing import List

class Solution:
    def returnToBoundaryCount(self, nums: List[int]) -> int:
        total_sum = 0
        count = 0
        
        for num in nums:
            total_sum += num
            if total_sum == 0:
                count += 1
                
        return count


sol = Solution()
output = sol.returnToBoundaryCount([-3, 3, -3])
print(output)



     
    

File: JavaScript/Math-Array-String/06.js
// 3028. Ant on the Boundary


/**
 * @param {number[]} nums
 * @return {number}
 */


var returnToBoundaryCount = function(nums) {
    let sum = 0, count = 0;
    for (let val of nums) {
        sum += val;
        if (sum === 0) {
            count++;
        }
    }
    return count;
};


console.log(returnToBoundaryCount([2,3,-5]));


// var returnToBoundaryCount = function(nums) {
//     let sum = 0, count = 0;
//     for (let i = 0; i < nums.length; i++) {
//         sum += nums[i];
//         if (sum === 0) {
//             count++;
//         }
//     }
//     return count;
// };
